### PR TITLE
CDC resync: inherit changes to mirror

### LINKED ISSUE #1819: Resync Mirror abandons added tables
When a mirror is resynced, it does not resync tables which were added to the mirror after creation. The new resynced mirror only has the original set of tables as part of its mapping

### PR DESCRIPTION
**Pull request by heavycrystal**
closes #1819

### CODE REVIEW

File: flow/shared/postgres.go [hunk 1]
```diff
@@ -104,3 +106,25 @@ func RollbackTx(tx pgx.Tx, logger log.Logger) {
...
 }
+
+func UpdateCDCConfigInCatalog(ctx context.Context, pool *pgxpool.Pool,
+	logger log.Logger, cfg *protos.FlowConnectionConfigs,
+) error {
+	logger.Info("syncing state to catalog: updating config_proto in flows", slog.String("flowName", cfg.FlowJobName))
+	var cfgBytes []byte
+	var err error
+
+	cfgBytes, err = proto.Marshal(cfg)
```
Comment (serprex) [inline flow/shared/postgres.go:117]:
```suggestion
	cfgBytes, err := proto.Marshal(cfg)
```

File: flow/workflows/cdc_flow.go [hunk 2]
```diff
@@ -91,69 +91,104 @@ func processCDCFlowConfigUpdate(
...
-		if len(flowConfigUpdate.AdditionalTables) == 0 {
-			return nil
-		}
-		if shared.AdditionalTablesHasOverlap(state.SyncFlowOptions.TableMappings, flowConfigUpdate.AdditionalTables) {
-			logger.Warn("duplicate source/destination tables found in additionalTables")
-			return nil
-		}
-		state.CurrentFlowStatus = protos.FlowStatus_STATUS_SNAPSHOT
+	defer func() {
+		syncStateToConfigProtoInCatalog(ctx, logger, cfg, state)
```
Comment (serprex) [inline flow/workflows/cdc_flow.go:95]: seems risky, should probably skip this if handling panic

File: flow/workflows/local_activities.go [hunk 3]
```diff
@@ -41,3 +45,11 @@ func getMaxSyncsPerCDCFlow(wCtx workflow.Context, logger log.Logger) uint32 {
 	}
 	return maxSyncsPerCDCFlow
 }
+
+func updateCDCConfigInCatalogActivity(ctx context.Context, logger log.Logger, cfg *protos.FlowConnectionConfigs) error {
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/workflows/local_activities.go:49]: can hold off if 1844 gets merged soon

File: flow/workflows/cdc_flow.go (hunk 2 above)
Comment (heavycrystal) [reply flow/workflows/cdc_flow.go:95]: added manually to all good return spots

**Review by Amogh-Bharadwaj — COMMENTED** [review]
functionally tested with resync to bq

**Pull request merged**